learning/bootstrap.py

```diff
@@ -139,43 +113,29 @@ async def teacher_loop(cfg: DictConfig, mle_log: MLELogger):
 
             # 2- Try to prove each of the conjectures
             examples = []
-            student_results= prove_conjectures(agent_dump, conjectures, theory, premises)
+            student_results = prove_conjectures(agent, conjectures, theory, premises)
 
             # 3- Train model on proofs and outcome of conjectures (easy, hard, timeout)
             # 3a- Look at all the success logprobs and compute the easy/hard threhsold.
-            success_logprobs = get_log_probs(student_results, i)
-            
+            success_logprobs = get_log_probs(student_results)
             ratio_proven = len(success_logprobs)/len(conjectures)
             log.info('%d out of %d conjectures proven. ratio = %f', 
                         len(success_logprobs), len(conjectures), ratio_proven)
 
-            if not success_logprobs:
+            if not success_logprobs and not cfg.skip_conjecturing:
                 log.warning('No solutions found in iteration %d - continuing to next iteration...', i)
                 continue
 
             # Add output of proving final goals to the list of proven conjectures
             student_results.extend(student_results_final)
 
-            thresholds = [np.percentile(success_logprobs, p)
-                          for _, p in difficulty_buckets]
-
-
-            log.debug('Thresholds: %s, min = %f, max = %f',
-                        list(zip([k for k, _ in difficulty_buckets], thresholds)),
-                        np.min(success_logprobs),
-                        np.max(success_logprobs))
-
-            hard_sol_log_probs = [logprob for logprob in success_logprobs if logprob >= thresholds[0]]
-            mean_hard_sol_log_prob = np.mean(hard_sol_log_probs) if hard_sol_log_probs else 0
+            mean_hard_sol_log_prob = sum(success_logprobs)/len(success_logprobs) if success_logprobs else 0
```

Comment:
```suggestion
            mean_hard_sol_log_prob = sum(success_logprobs)/len(success_logprobs)
```
success_logprobs is never None since we skip the iteration otherwise (see above)

Reply:
nvm